src/main/java/com/arpnetworking/metrics/mad/Bucket.java

```diff
@@ -349,6 +361,7 @@ private void addMetric(
     }
 
     private boolean _isOpen = true;
+    private ZonedDateTime _min_request_time = null;
```

Comment:
Prefer `Optional` to `null`, or as you are mutating it at least mark it `@Nullable`.

Reply:
Made it `Optional`! Should I have made `DefaultRecord._request_time` an `Optional` as well?

It's not clear to me when I should use `null` vs. `Optional`.